### DISCUSSION TITLE
linking issue: fatal error: ParameterObject.h: No such file or directory

### DISCUSSION THREAD

**Discussion by Hassan-Bahrami** [InteractiveComputerGraphics/SPlisHSPlasH#222] [Q&A] [answered]
Hi team,

Thanks for the library. I am struggling linking SPlishSplasH to my project for a few days. Finally I could link it with following Cmake commands:

```
target_include_directories(${PROJECT_NAME} PUBLIC ../extern/SPlisHSPlasH)

target_link_directories(${PROJECT_NAME} PRIVATE ../extern/SPlisHSPlasH/build/lib)

target_link_libraries(${PROJECT_NAME} CompactNSearch igl::glfw ${OPENGL_LIBRARY} ${OpenGL_libs} SPlisHSPlasH libMD5 libimgui libpartio libSimulatorBase libtinyexpr libUtilities libzlib)

```

Pleas note that I have already linked OpenGL libraries, LibIgl and CompactNSearch separately (they work without error). When I want to build my project, cmake command works without error. However, for making I get following error:

```
../extern/SPlisHSPlasH/SPlisHSPlasH/FluidModel.h:9:10: fatal error: ParameterObject.h: No such file or directory
    9 | #include "ParameterObject.h"
```
I searched in the SPlishSplasH directory. There is no "ParameterObject.h" file. I was wondering something is missed in the library or I made a mistake somewhere?

**Comment by janbender** [comment, marked as answer]
As mentioned in the readme file, the project depends on our project: 

https://github.com/InteractiveComputerGraphics/GenericParameters

which is a header-only project. So there is no lib. You can see here: 

https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/blob/master/CMake/SetUpExternalProjects.cmake#L32

how the project is included in SPlisHSPlasH.

Hope that helps!

**Reply by Hassan-Bahrami** [reply]
I resolved the issue by linking "GenericParameters" lib.

Thanks
